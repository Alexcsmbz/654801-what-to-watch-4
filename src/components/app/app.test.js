import React from 'react';
import renderer from 'react-test-renderer';
import App from './app.jsx';

const name = `Whatever film`;
const genre = `Criminal`;
const reliseDate = `1999`;
const films = [
  {
    name: `Film 1`,
    img: `img/img-1.jpg`,
  },
  {
    name: `Film 2`,
    img: `img/img-2.jpg`,
  },
  {
    name: `Film 3`,
    img: `img/img-3.jpg`,
  },
  {
    name: `Film 4`,
    img: `img/img-4.jpg`,
  },
];

describe(`App snapshot test`, () => {
  it(`App should render Main component with films name, genre, relise date, films list`, () => {
    const tree = renderer.create(
        <App
          name={name}
          genre={genre}
          reliseDate={reliseDate}
          films={films}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
