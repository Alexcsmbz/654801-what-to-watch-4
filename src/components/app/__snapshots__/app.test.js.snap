// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test App should render Main component with movies name, genre, relise date, movies list 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Whatever movie
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Criminal
            </span>
            <span
              className="movie-card__year"
            >
              1999
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <button
    onClick={[Function]}
  >
    onFilterClick
  </button>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active "
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            All genres
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Comedies
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Crime
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Documentary
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Dramas
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Horror
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Kids & Family
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Romance
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Sci-Fi
          </div>
        </li>
        <li
          className="catalog__genres-item undefined"
          onClick={[Function]}
        >
          <div
            className="catalog__genres-link"
          >
            Thrillers
          </div>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-1"
              onClick={[Function]}
            >
              movie 1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-2"
              onClick={[Function]}
            >
              movie 2
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-3"
              onClick={[Function]}
            >
              movie 3
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            controls={true}
            height="175"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            width="280"
          >
            <source
              type="video/webm"
            />
            <source
              type="video/mp4"
            />
          </video>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movie-page/movie-4"
              onClick={[Function]}
            >
              movie 4
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
